12-http/example3/main.go

```diff
@@ -39,7 +68,9 @@ func usersHandler(w http.ResponseWriter, r *http.Request) {
 	switch r.Method {
 	case "GET", "HEAD":
 		// List the users
+		mu.RLock()
```

Comment:
Don't forget the old saying: "Lock data, not code"

I'll add the anonymous block, but we need to make it extra clear that critical sections are an unfortunate side-effect of how locks work -- they're not the purpose of locks. High level mutex semantics in Go, like in many other C-derived languages, are a mental construct only (on the other hand, channels have language-defined high level semantics, which makes them safer and more powerful when used correctly).